components_library/boards/lupv_board.py

```diff
@@ -403,6 +423,22 @@ def generate_device_tree(self, outdir: str) -> None:
         lupio_pic_node.appendCompatible(["lupio,pic"])
         soc_node.append(lupio_pic_node)
 
+        # LupioIPI Device
+        lupio_ipi = self.lupio_ipi
+        lupio_ipi_node = lupio_ipi.generateBasicPioDeviceNode(soc_state,
+                                "lupio-ipi", lupio_ipi.pio_addr,
+                                lupio_ipi.pio_size)
+        int_extended = list()
+        for i, core in enumerate(self.get_processor().get_cores()):
+            phandle = state.phandle(f"cpu@{i}.int_state")
+            int_extended.append(phandle)
+            int_extended.append(0x1)    #IRQ_S_SOFT
```

Comment:
Use named variable instead of hardcoded value if possible.

Here you have the typical case of what I was mentioning on Teams yesterday. Your device tree will be correct, but is not describing the actual platform (since your code currently connects the IPI to the IRQ_M_SOFT pins).